02-Generics/Generics/Generics.cs

```diff
@@ -105,17 +177,35 @@ public static class ArrayExtentions {
 	///   This code should return the same MyService object every time:
 	///   MyService singleton = Singleton<MyService>.Instance;
 	/// </example>
-	public static class Singleton<T> {
-		// TODO : Implement generic singleton class 
+	public static class Singleton<T> where T : class, new()
+	{
+		static Singleton()
+		{
+
+		}
+
+		class SingletonCreator
```

Comment:
I think you don't need this class. Replace on private static T instance = new T();
and use this variable in public static T Instance property